compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2614,7 +2614,14 @@ public void visit(BLangQueryExpr queryExpr) {
         for (WhereClauseNode whereClauseNode : whereClauseList) {
             whereEnv = typeCheckWhereClause((BLangWhereClause) whereClauseNode, selectClause, parentEnv);
         }
-        checkExpr(selectClause.expression, whereEnv);
+
+        if (queryExpr.expectedType.tag != symTable.noType.tag) {
```

Comment:
I guess we can use the `expType` instead.
```suggestion
        if (expType.tag != symTable.noType.tag) {
```

Reply:
Shall we check 
```suggestion
        if (expType.tag == TypeTags.ARRAY) {
```
instead?